packages/client/src/forms/register/mappings/query/registration-mappings.ts

```diff
@@ -168,26 +175,30 @@ export const userTransformer =
           !action && regStatus && statuses.includes(regStatus)
       )
 
-    if (history) {
-      const district = history.location?.id
-        ? offlineData?.locations?.[history.location.id]
-        : null
-      const state = district
-        ? offlineData?.locations?.[district.partOf.split('/')[1]]
-        : null
-      const province = state
-        ? offlineData?.locations?.[state.partOf.split('/')[1]]
-        : null
+    if (history?.location && offlineData) {
+      const { country, ...locationHierarchyIds } = getLocationHierarchy(
+        history.location.id,
+        offlineData.locations
+      )
+      const locationHierarchy: Record<
+        string,
+        string | AdminStructure | undefined
+      > = { country }
+
+      for (const [key, value] of Object.entries(locationHierarchyIds)) {
+        locationHierarchy[`${key}Id`] = value
+        locationHierarchy[key] = offlineData.locations[value]
+      }
+
       transformedData[targetSectionId || sectionId][
         targetFieldName || 'registrar'
       ] = {
         name: getUserFullName(history),
         role: getUserRole(history),
         office: history.office,
+        officeId: history.office?.id,
         date: history.date,
-        district,
-        state,
-        province,
```

Comment:
This is the only handlebar I removed, which was not a valid one in the first place.